src/snovault/tests/test_schema_utils.py

```diff
@@ -32,3 +32,39 @@ def test_uniqueItems_validates_normalized_links(content, threadlocals):
         errors[0].message == "['{}', '{}'] has non-unique elements".format(
             uuid, uuid)
     )
+
+
+def test_pattern_validates_strings_or_array_of_strings(content, threadlocals):
+    schema = {
+        'items': {
+            'pattern': 'option1|option2'
+        }
+    }
+
+    uuid = 'option2'
+    data = [
+        uuid,
+        '/testing-link-targets/{}'.format(uuid),
+    ]
+
+    validated, errors = validate(schema, data)
+    assert len(errors) == 0
+
+    schema['items']['pattern'] = ['option1|', 'option2']
```

Comment:
You probably already thought about just joining on `|` to clean up formatting a bit.

Reply:
Yes, that's actually the use case we have. But after options, the last line in our use case is an ID logic for a suffix. We can't join by '|' 😢